src/wallet/api/wallet.cpp

```diff
@@ -197,6 +197,44 @@ bool Wallet::addressValid(const std::string &str, bool testnet)
   return get_account_integrated_address_from_str(address, has_payment_id, pid, testnet, str);
 }
 
+bool Wallet::keyValid(const std::string &key_string, const std::string &address_string, bool isViewKey, bool testnet, std::string &error)
+{
+  bool has_payment_id;
+  cryptonote::account_public_address address;
+  crypto::hash8 pid;
+  if(!get_account_integrated_address_from_str(address, has_payment_id, pid, testnet, address_string)) {
+      error = tr("Failed to parse address");
+      return false;
+  }
+  
+  cryptonote::blobdata key_data;
+  if(!epee::string_tools::parse_hexstr_to_binbuff(key_string, key_data))
+  {
+      error = tr("Failed to parse key");
+      return false;
+  }
```

Comment:
Need to check that you actually got the right amount of data, the hex string might be the wrong size, and might segfault below.